infra/app.tf

```diff
@@ -0,0 +1,411 @@
+  description       = "Allow all egress"
+}
+
+
+resource "aws_cloudwatch_log_group" "this" {
+  name = var.name_prefix
+}
+
+resource "aws_ecs_cluster" "this" {
+  name = var.name_prefix
+}
+
+locals {
+  app_environment = [
+    {
+      name  = "HOST"
+      value = "${var.host}"
+    },
+    {
+      name  = "WCA_HOST"
+      value = "${var.wca_host}"
+    }
+  ]
+}
+
+#resource "aws_ecs_task_definition" "migrate" {
+#  family = "${var.name_prefix}-migrate"
+#
+#  network_mode             = "awsvpc"
+#  requires_compatibilities = ["EC2"]
+#
+#  cpu    = 512
+#  memory = 512
+#
+#  container_definitions = jsonencode([
+#    {
+#      name         = "main"
+#      image        = "${aws_ecr_repository.this.repository_url}:latest"
+#      command      = ["/app/bin/migrate"]
+#      portMappings = []
+#      logConfiguration = {
+#        logDriver = "awslogs"
+#        options = {
+#          awslogs-group         = "${aws_cloudwatch_log_group.this.name}"
+#          awslogs-region        = "${var.region}"
+#          awslogs-stream-prefix = "${var.name_prefix}"
+#        }
+#      }
+#      environment = local.app_environment
+#    }
+#  ])
+#
+#  tags = {
+#    Name = "${var.name_prefix}-migrate"
+#  }
+#}
+
+data "aws_iam_policy_document" "task_assume_role_policy" {
+  statement {
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "task_execution_role" {
+  name               = "${var.name_prefix}-task-execution-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "task_execution_role_attachment" {
+  role       = aws_iam_role.task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "task_role" {
+  name               = "${var.name_prefix}-task-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume_role_policy.json
+}
+
+data "aws_iam_policy_document" "task_policy" {
+  statement {
+    actions = [
+      "ssmmessages:CreateControlChannel",
+      "ssmmessages:CreateDataChannel",
+      "ssmmessages:OpenControlChannel",
+      "ssmmessages:OpenDataChannel",
+    ]
+
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role_policy" "task_policy" {
+  role   = aws_iam_role.task_role.name
+  policy = data.aws_iam_policy_document.task_policy.json
+}
+
+resource "aws_ecs_task_definition" "main" {
+  family = var.name_prefix
+
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["EC2"]
+
+  # We configure the roles to allow `aws ecs execute-command` into a task,
+  # as in https://aws.amazon.com/blogs/containers/new-using-amazon-ecs-exec-access-your-containers-fargate-ec2
+  execution_role_arn = aws_iam_role.task_execution_role.arn
+  task_role_arn      = aws_iam_role.task_role.arn
+
+  cpu = "1024"
+  memory = "1024"
```

Comment:
I think if the instance itself has `2048` then it may not fit two containers, because the instance itself also uses a small bit of memory. Not sure what's the optimal value (800-900?), but just pointing out that blue/green would likely spawn a separate instance.

Reply:
makes sense, these values are chosen at random now anyway. I created an issue here #26 to revisit once we have something to load test.